seismic/xcorqc/correlator.py

```diff
@@ -53,162 +53,91 @@ def xyz2rtp(x, y, z):
-                            zchannels.add(ch.code)
-                        if 'N' in ch.code or 'n' in ch.code:
-                            nchannels.add(ch.code)
-                        if 'E' in ch.code or 'e' in ch.code:
-                            echannels.add(ch.code)
-                    # end for
-                # end for
-            # end for
+                         np.radians(90 - self.metadata[netsta][1]),
+                         np.radians(self.metadata[netsta][0])])
         # end for
 
         rtps = np.array(rtps)
         xyzs = rtp2xyz(rtps[:, 0], rtps[:, 1], rtps[:, 2])
 
         self._tree = cKDTree(xyzs)
         self._cart_location = defaultdict(list)
-        for i, s in enumerate(self.stations):
-            self._cart_location[s] = xyzs[i, :]
+        for i, ns in enumerate(self.netsta_list):
+            self._cart_location[ns] = xyzs[i, :]
         # end for
-
-        if len(zchannels):
-            if(zchan is None):
-                assert len(zchannels) == 1, 'Multiple z-channels found %s' % (zchannels)
-            else:
-                zchannels = set([zchan]) if zchan in zchannels else None
-            if(zchannels is None):
-                assert 0, 'Invalid z-channel name'
-
-        if len(nchannels):
-            if(nchan is None):
-                assert len(nchannels) == 1, 'Multiple n-channels found %s' % (nchannels)
-            else:
-                nchannels = set([nchan]) if nchan in nchannels else None
-            if(nchannels is None):
-                assert 0, 'Invalid n-channel name'
-
-        if len(echannels):
-            if(echan is None):
-                assert len(echannels) == 1, 'Multiple e-channels found %s' % (echannels)
-            else:
-                echannels = set([echan]) if echan in echannels else None
-            if(echannels is None):
-                assert 0, 'Invalid e-channel name'
-
-        self.zchannel = zchannels.pop() if len(zchannels) else None
-        self.nchannel = nchannels.pop() if len(nchannels) else None
-        self.echannel = echannels.pop() if len(echannels) else None
     # end func
 
-    def get_closest_stations(self, station_name, other_dataset, nn=1):
-        assert isinstance(station_name, basestring), 'station_name must be a string'
+    def get_closest_stations(self, netsta, other_dataset, nn=1):
+        assert isinstance(netsta, basestring), 'station_name must be a string'
         assert isinstance(other_dataset, Dataset), 'other_dataset must be an instance of Dataset'
-        station_name = station_name.upper()
+        netsta = netsta.upper()
 
-        assert station_name in self.stations, 'station %s not found'%(station_name)
+        assert netsta in self.netsta_list, '%s not found'%(netsta)
 
-        d, l = other_dataset._tree.query(self._cart_location[station_name], nn)
+        d, l = other_dataset._tree.query(self._cart_location[netsta], nn)
 
         if isinstance(l, int):
```

Comment:
I think this block of code up to line 106 relies on behavior deprecated in Python 3, and is a little obtuse. I got it working with Python 3 in another branch and it looked like this:

```
        if np.isscalar(d):
            d = np.array([d])
        if np.isscalar(l):
            l = np.array([l])
        l = l[np.isfinite(d)]
        l = np.unique(l)
```

This code should be double checked at runtime if you use it, as there is no test suite.

Reply:
Thanks @medlin01GA. I hope to do the py3 conversion once we have tests in place. As a part of the documentation process I'll be adding code comments as well. I agree, this snippet is a little obtuse. What we're doing here is as follows:

* Fetch N nearest neighbours from another `DataSet`. However, if the other `DataSet` contains < N stations, the missing neighbours will be marked by N in the return value `l`, which can be a tuple or a scalar.
* Make sure `l` is a list (line 100)
* Apply a filter to get rid of missing neighbours, as described above (line 103)
* Make sure `l` is a list (line 105)